addons/web_editor/static/tests/html_field_tests.js

```diff
@@ -36,13 +124,199 @@ QUnit.module("WebEditor.HtmlField", ({ beforeEach }) => {
+            '<p>toto toto toto</p><p>tata</p>',
+            "should have rendered a div with correct content");
+        assert.strictEqual(field.getAttribute('style'), 'height: 100px',
+            "should have applied the style correctly");
+    });
+
+    QUnit.test('check if required field is set', async function (assert) {
+        assert.expect(3);
+
+        serverData.models['note.note'].records = [{
+            id: 1,
+            display_name: "first record",
+            header: "",
+            body: "<p>toto toto toto</p><p>tata</p>",
+        }];
+
+        await makeView({
+            type: "form",
+            resId: 1,
+            resModel: "note.note",
+            serverData,
+            arch: `
+                <form>
+                    <field name="header" widget="html" style="height: 100px"/>
+                </form>`,
+        });
+        await click(target, '.o_form_button_save');
+        assert.containsOnce(target, ".o_notification");
+        const notif = target.querySelector(".o_notification");
+        assert.strictEqual(
+            notif.querySelector(".o_notification_content").textContent,
+            "Header"
+        );
+        assert.hasClass(notif, "border-danger");
+
+    });
+
+    QUnit.test('colorpicker', async function (assert) {
+        assert.expect(10);
+
+        serverData.models['note.note'].records = [{
+            id: 1,
+            display_name: "first record",
+            header: "<p>  &nbsp;&nbsp;  <br>   </p>",
+            body: "<p>toto toto toto</p><p>tata</p>",
+        }];
+
+        // Add the ajax service (legacy), because wysiwyg RPCs use it.
+        patchWithCleanup(legacyEnv, {
+            services: {
+                ...legacyEnv.services,
+                ajax: {
+                    rpc: async (route, args) => {
+                        if (args.model === "ir.ui.view" && args.method === 'render_public_asset') {
+                            if (args.args[0] === "web_editor.colorpicker") {
+                                return COLOR_PICKER_TEMPLATE;
+                            }
+                            if (args.args[0] === "web_editor.snippets") {
+                                return SNIPPETS_TEMPLATE;
+                            }
+                        }
+                    },
+                },
+            }
+        });
+
+        await makeView({
+            type: "form",
+            resId: 1,
+            resModel: "note.note",
+            serverData,
+            arch: `
+                <form>
+                    <field name="body" widget="html" style="height: 100px"/>
+                </form>`,
+        });
+
+        const field = target.querySelector('.o_field_html[name="body"]');
+        const editable = field.querySelector("[contenteditable='true']");
+        await click(editable);
+
+        assert.strictEqual(
+            window.getComputedStyle(document.querySelector("#toolbar")).getPropertyValue("visibility"),
```

Comment:
This is very long so I would have put at list part of it into a variable. Maybe the toolbar can be a variable ? Because we're query selecting it multiple times.

Reply:
ok